client/lib/wp/localization/index.js

```diff
@@ -16,7 +16,7 @@ export function addLocaleQueryParam( params ) {
 	}
 
 	let localeQueryParam;
-	const query = parse( params.query );
+	const query = parse( params.query, { depth: 10 } );
```

Comment:
Is a depth of 10 necessary?

Reply:
The depth of the current query is 6, but I wanted to avoid running into the same problem if we changed it in the future. 

Given that this is the first occurrence of this issue and that the object we'd like to parse aren't big, I thought it's safe to give ourselves some extra headroom.